### ISSUE TITLE
Week 1 Instagram Client

### ISSUE THREAD

**Issue by maygupta** [maygupta/MyInstagram#1]
My app has basic features ready but need some UI polishing and some advanced features.
/cc @codepathreview @codepath

My question is should we submit the issue as soon as we have basic feature working like we did for Todo app and as and when we add new feature we add that as a comment or should we notify Codepath team at the very end when every thing is ready from our end ?

**Comment by nidhi1608** [comment]
You can notify us when you're ready for us to review the entire app.

**Comment by maygupta** [comment]
Ok Thanks @nidhi1608 

One question,  
"Advanced: Display each photo with the same style and proportions as the real Instagram (see screens below)"

But isn't this already done by Picaso. I didn't do any scaling explicitly and the app renders the images with proper height and width without any stretching. I have added a demo in README just for the reference.

**Comment by nidhi1608** [comment]
Yes, you can ignore that user story. I'll update the assignment.

**Comment by maygupta** [comment]
@nidhi1608 

My app is ready for the review. I have skipped one of the item (to show the Video for video posts) since I am not sure if I will be able to take out time to work on that. If I do get time I will surely add that too. But I thought posting this early would be good for code review.

**Comment by maygupta** [comment]
@nidhi1608 
Hi Nidhi,
Just pinging again for your code review comments.
Thanks,
Mayank

**Comment by nidhi1608** [comment]
Mayank, we'll be reviewing it tonight after the deadline.
- Nidhi

**Comment by codepathreview** [comment]
:+1: Excellent work overall, good job getting most of the optional stories done. A few notes after checking out the code:
- Good layout, proper view naming, usage of sp/dp, margin and relative placement rules
- Consider externalizing string resources for internationalization, even if they're just hint texts. Consider using `android:hint` instead of `android:text` as appropriate.
- Nice to see you used the `SwipeRefreshLayout` for easy stream refreshes
- Consider [better organizing your source files](http://guides.codepath.com/android/Organizing-your-Source-Files)
- Good job using the [ViewHolder pattern](https://github.com/codepath/android_guides/wiki/Using-an-ArrayAdapter-with-ListView#improving-performance-with-the-viewholder-pattern) to improve performance of photos `ListView`. Consider adding the same for `PhotoAdapter`
- Good job abbreviating timestamps. Consider externalizing it to a Utils class.
- Consider including a [check to see if internet is available](https://guides.codepath.com/android/Sending-and-Managing-Network-Requests) to detect any network failures. Show an appropriate message in case of failure.
- Consider overriding `onFailure` method in `JsonHttpResponseHandler` to handle error messages appropriately.
- Consider [externalizing](https://guides.codepath.com/android/Converting-JSON-to-Models) the JSON parsing logic to the model class

One of the most important part of these projects is that you add additional features and tweak the UI / UX because that will provide many more learning opportunities. I would encourage you to complete the projects each week with required stories early and then spend time adding your own UI elements and experimenting with optional extensions that will improve the user experience.

We have provided a detailed [Project 1 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_1_feedback) which covers the most common points we see for this project. Read through the [feedback guide point-by-point](http://courses.codepath.com/snippets/intro_to_android/project_1_feedback) to determine other ways you could improve your submission. You should consider going back and implementing applicable feedback as well. Keep in mind that one of the most important parts of Android development is learning the correct patterns and conventions.

Hopefully this first project has given you a better sense of working with RelativeLayout which is a very flexible layout system, probably one of the most powerful responsive-first layout systems available across web and mobile platforms. This assignment also gave us our first introduction to networking, working with APIs and loading remote images. The next assignment will introduce new concept such as accepting user input and navigating between activities but will also reinforce important concepts such as networking, using APIs, handling remote images.

If you have any particular questions about the assignment in general or on any of the feedback, feel free to reply here or or email us support@codepath.com.

**P.S.** Good to see you properly added the readme with features and screenshots to your project as described in the [submitting assignments](http://courses.codepath.com/courses/intro_to_android/pages/submitting_assignments) guide!

**Comment by maygupta** [comment]
Thanks for the detailed feedback. Its good to know what could be done better. Will work on these comments
